src/joystick/linux/SDL_sysjoystick.c

```diff
@@ -1333,7 +1573,28 @@ static int LINUX_JoystickSendEffect(SDL_Joystick *joystick, const void *data, in
 
 static int LINUX_JoystickSetSensorsEnabled(SDL_Joystick *joystick, SDL_bool enabled)
 {
-    return SDL_Unsupported();
+    if (!joystick->hwdata->has_accelerometer && !joystick->hwdata->has_gyro) {
+        return SDL_Unsupported();
+    }
+    if (enabled == joystick->hwdata->report_sensor) {
+        return 0;
+    }
+
+    if (enabled) {
+        SDL_assert(joystick->hwdata->item_sensor);
```

Comment:
Reviewing my own work again.
If we call `LINUX_JoystickSetSensorsEnabled()` after the joystick is unplugged then this assertion will trigger. `item_sensor` is set to NULL by `RemoveSensorlistItem()`.

Reply:
PR #8357 if you want to integrate this in here.

Reply:
If @slouken approves #8357 and merges it, I'll push that change to this branch/PR too. Thanks!

Reply:
Included #8357 in 503a24592da59e640ed067850af748936809149f